### PR TITLE
Add Google Cloud Storage (GCS) backend 

### PR DESCRIPTION
**Pull request by haimjether**
* Add extra google-cloud-storage requirements

* Add gcs backend module

* Add gcs backend to userguide configuration

* Add gcs to backends in README

* Add gcs app Namespace

* Add configuration documentation

*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryq.dev/en/main/contributing.html).

### CODE REVIEW

**Review by thedrow — APPROVED** [review]

File: celery/app/backends.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
     'azureblockblob': 'celery.backends.azureblockblob:AzureBlockBlobBackend',
     'arangodb': 'celery.backends.arangodb:ArangoDbBackend',
     's3': 'celery.backends.s3:S3Backend',
+    'gs': 'celery.backends.gcs:GCSBackend',
```
Comment (jayaddison) [inline celery/app/backends.py:37]: @haimjether @thedrow something I was wondering about during upgrade-review for `celery` - is this meant to be registered as `gs` or is that a typo of `gcs`?

Comment (haimjether) [reply celery/app/backends.py:37]: @jayaddison , it was meant to be registered as `gs` , to mimic the blobs path prefix on GCS. E.g 
`gs://my-bucket/my-folder` 

IMHO this is similar to AWS `s3` one.

Comment (jayaddison) [reply celery/app/backends.py:37]: :+1: great, glad to know it's as-intended.  Thanks @haimjether!

**Review by auvipy — COMMENTED** [review]
this misses the .. versionadded:: 5.4 annotation

**Pull request merged**